Missing constructors for AbstractTableConstraint

Maybe it's just too late, but I cannot build the current Hyrise **main** (e4a6d59cee76496b8572f738aa7c06eb6b0a67d3) due to the following problem:
```
../src/lib/storage/constraints/abstract_table_constraint.hpp:18:11: error: definition of implicit copy assignment operator for 'AbstractTableConstraint' is deprecated because it has a user-declared destructor [-Werror,-Wdeprecated]
  virtual ~AbstractTableConstraint() = default;
```

--- a/src/lib/storage/constraints/abstract_table_constraint.hpp
+++ b/src/lib/storage/constraints/abstract_table_constraint.hpp
@@ -14,6 +14,12 @@ namespace opossum {
 class AbstractTableConstraint {
  public:
   explicit AbstractTableConstraint(std::unordered_set<ColumnID> init_columns);
+
+  AbstractTableConstraint(const AbstractTableConstraint&) = default;
+  AbstractTableConstraint(AbstractTableConstraint&&) = default;
+  AbstractTableConstraint& operator=(const AbstractTableConstraint&) = default;
+  AbstractTableConstraint& operator=(AbstractTableConstraint&&) = default;
+
   virtual ~AbstractTableConstraint() = default;
 
   const std::unordered_set<ColumnID>& columns() const;

